server.go

```diff
@@ -332,6 +335,7 @@ type Server struct {
 	// ConnState specifies an optional callback function that is
 	// called when a client connection changes state. See the
 	// ConnState type and associated constants for details.
+	// Deprecated: Use <Trace> instead.
```

Comment:
if we want to deprecate this in favor of ServerTrace, we should provide an identical feature. I don't feel like
```go
s.trace := &fasthttp. ServerTrace{
	GotConn:	connStateChanged,
	ClosedConn:	connStateChanged,
	ActivatedConn:	connStateChanged,
	IdledConn:	connStateChanged,
	HijackedConn:	connStateChanged,
}
```

instead of

```go
s.ConnState = connStateChanged
```

Reply:
Having separate hooks for the various state transitions simplifies the code that the user of `fasthttp` has to write.

In your example, `connStateChanged` might look like this:
```
func handleNewConnection(conn net.Conn) {
	logger.Info("new connection established", zap.String("remote_addr", conn.RemoteAddr().String()))
	// do other things
}

func handleClosedConnection(conn net.Conn) {
	logger.Info("connection closed", zap.String("remote_addr", conn.RemoteAddr().String()))
	// do other things
}

func connStateChanged(conn net.Conn, state ConnState) {
	switch state {
	case fasthttp.StateNew:
		handleNewConnection(conn)
	case fasthttp.StateClosed:
		handleClosedConnection(conn)
	// ...
	}
}
```

Where as using the new `ServerTrace` implementation, this code will become:
```
func handleNewConnection(conn net.Conn) {
	logger.Info("new connection established", zap.String("remote_addr", conn.RemoteAddr().String()))
	// do other things
}

func handleClosedConnection(conn net.Conn) {
	logger.Info("connection closed", zap.String("remote_addr", conn.RemoteAddr().String()))
	// do other things
}

s.Trace = fasthttp.ServerTrace{
	GotConn: handleNewConnection,
	ClosedConn: handleClosedConnection,
	// ...
}
```

As a user of `fasthttp` I personally would prefer that an improvement as I no longer have to deal with differentiating the various states.